### DISCUSSION TITLE
CTRL-C gets CancelledError. Is this a uvicorn issue?

### DISCUSSION THREAD

**Discussion by Dronakurl** [fastapi/fastapi#11514] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
async def example():
    return {"message": "Adid"}

if __name__ == "__main__":
    uvicorn.run("bb:app", host="0.0.0.0", port=8000, reload=False, workers=1, log_level="warning")
```

### Description

When I press CTRL-C, the server does not shutdown gracefully, but produces a "CancelledError".

### Operating System

Linux

### Operating System Details

python3.12

### FastAPI Version

0.109.2

### Pydantic Version

2.7.1

### Python Version

3.12.2

**Comment by Kludex** [comment]
Is it only happening on the last version?

**Reply by Dronakurl** [reply]
No, I use version "0.109.2". But I just started a fresh install of only fastapi in a new virtual environment with the 0.110.3 and I got the same issue. uvicorn is version 0.29.0.
The problem dissappears when I set `reload=True`

**Reply by Kludex** [reply]
Ah, we are on FastAPI repo... I meant, "Is it only happening on the last uvicorn version?"

**Reply by Dronakurl** [reply]
Ah, i see, using uvicorn 0.28.0, the problem is gone. But strangely it is not present when starting uvicorn from the command line: `uvicorn bb:app`

**Reply by LordLumineer** [reply]
the error is also present with ctrl+s, but unfortunately with the latest version 0.111.0 it brings fastapi-cli (0.0.2) which requires uvicorn 0.29.0. 

```bash
INFO:     Will watch for changes in these directories: ['D:\\Desktop\\Codding Space\\EventKit\\auth\\server']
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [15496] using WatchFiles
INFO:     Started server process [20468]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
WARNING:  WatchFiles detected changes in 'main.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [20468]
Process SpawnProcess-1:
Traceback (most recent call last):
  File "c:\Users\Lumineer\AppData\Local\Programs\Python\Python312\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Lumineer\AppData\Local\Programs\Python\Python312\Lib\asyncio\base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Lumineer\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
  File "c:\Users\Lumineer\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "d:\Desktop\Codding Space\EventKit\auth\server\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "d:\Desktop\Codding Space\EventKit\auth\server\.venv\Lib\site-packages\uvicorn\server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Lumineer\AppData\Local\Programs\Python\Python312\Lib\asyncio\runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "c:\Users\Lumineer\AppData\Local\Programs\Python\Python312\Lib\asyncio\runners.py", line 123, in run
    raise KeyboardInterrupt()
KeyboardInterrupt
INFO:     Started server process [29732]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
```

**Reply by Dronakurl** [reply]
Should this be an issue then? It is not serious, for sure.

**Comment by YuriiMotov** [comment]
As far as I know that was fixed. So, I'll close this.
If you still have this problem, let us know
